blockchain/miner.go

```diff
@@ -0,0 +1,103 @@
+package blockchain
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"math"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	// MiningHeaderLen is the length of the MiningHeader struct in bytes
+	MiningHeaderLen = (3 * HashLen) + (1 * (32 / 8)) + (1 * (64 / 8))
+)
+
+// MiningHeader contains the metadata required for mining
+type MiningHeader struct {
+	LastBlock Hash
+	RootHash  Hash
+	Target    Hash
+	Time      uint32
+	Nonce     uint64
+}
+
+// Marshal converts a Mining Header to a byte slice
+func (mh *MiningHeader) Marshal() []byte {
+	var buf []byte
+	buf = append(buf, mh.LastBlock.Marshal()...)
+	buf = append(buf, mh.RootHash.Marshal()...)
+	buf = append(buf, mh.Target.Marshal()...)
+	AppendUint32ToSlice(&buf, mh.Time)
+	AppendUint64ToSlice(&buf, mh.Nonce)
+	return buf
+}
+
+// SetMiningHeader sets the mining header
+func (mh *MiningHeader) SetMiningHeader(lastBlock Hash, rootHash Hash, target Hash) {
```

Comment:
it sets the time to the time variable to the current time in the miningheader struct, not quite sure how to set default values in golang structs